lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -135,14 +135,7 @@ protected function doLoadMetadata($class, $parent, $rootEntityFound, array $nonS
             throw MappingException::reflectionFailure($class->getName(), $e);
         }
 
-        // If this class has a parent the id generator strategy is inherited.
-        // However this is only true if the hierarchy of parents contains the root entity,
-        // if it consists of mapped superclasses these don't necessarily include the id field.
-        if ($parent && $rootEntityFound) {
-            $this->inheritIdGeneratorMapping($class, $parent);
-        } else {
-            $this->completeIdGeneratorMapping($class);
-        }
+        $this->completeIdGeneratorMapping($class);
```

Comment:
This only fills in missing defaults; probably not necessary once we've got all this in the base class and inherit it. 

I don't know if it is worth coming up with fancy rules here when we may skip this step, since in the end, all metadata is cached, right?